### PR TITLE
Feature/csharp matrix formats

### PR DESCRIPTION
**Pull request by ZEROM22**
added work with matrix fomats

### CODE REVIEW

File: web/documentserver-example/csharp/Default.aspx.cs [hunk 1]
```diff
@@ -284,9 +244,9 @@ public static string DocumentType(string fileName)
         {
             var ext = Path.GetExtension(fileName).ToLower();
 
-            if (FileType.ExtsDocument.Contains(ext)) return "word";  // word for text document extensions
-            if (FileType.ExtsSpreadsheet.Contains(ext)) return "cell";  // cell for spreadsheet extensions
-            if (FileType.ExtsPresentation.Contains(ext)) return "slide";  // slide for presentation extensions
+            if (FormatManager.DocumentExtensions().Contains(ext)) return "word";  // word for text document extensions
+            if (FormatManager.SpreadsheetExtensions().Contains(ext)) return "cell";  // cell for spreadsheet extensions
+            if (FormatManager.PresentationExtensions().Contains(ext)) return "slide";  // slide for presentation extensions
```
Comment (aleksandrfedorov97) [inline web/documentserver-example/csharp/Default.aspx.cs:249]: Why do you call the All() method 3 times? Therefore the file "onlyoffice-docs-formats.json" is read 3 times.

File: web/documentserver-example/csharp/FormatManager.cs [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        }
+
+        public static List<string> AllExtensions()
+        {
+            return All()
+                .Select(format => format.Extension())
+                .ToList();
+        }
+
+        public static List<Format> All()
```
Comment (aleksandrfedorov97) [inline web/documentserver-example/csharp/FormatManager.cs:160]: Maybe initialize the formats array when creating an instance of the class?

**Review by aleksandrfedorov97 — CHANGES_REQUESTED** [review]

File: web/documentserver-example/csharp/FormatManager.cs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                throw new FileNotFoundException("The JSON file does not exist.");
+            }
+        }
+
+        private static string GetDirectory()
+        {
+            string directory = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "assets", "document-formats");
+            return Path.GetFullPath(directory);
+        }
+    }
```
Comment (sshakndr) [inline web/documentserver-example/csharp/FormatManager.cs:187]: Do you really have to use 2 functions? Like for example `FillableExtensions()` and `Fillable()`.

**Review by sshakndr — APPROVED** [review]

File: web/documentserver-example/csharp/Default.aspx.cs (hunk 1 above)
Comment (ZEROM22) [reply web/documentserver-example/csharp/Default.aspx.cs:249]: > Why do you call the All() method 3 times? Therefore the file "onlyoffice-docs-formats.json" is read 3 times.

I added format caching

File: web/documentserver-example/csharp/FormatManager.cs (hunk 2 above)
Comment (ZEROM22) [reply web/documentserver-example/csharp/FormatManager.cs:162]: > Maybe initialize the formats array when creating an instance of the class?

I added format caching

File: web/documentserver-example/csharp/FormatManager.cs (hunk 3 above)
Comment (ZEROM22) [reply web/documentserver-example/csharp/FormatManager.cs:193]: > Do you really have to use 2 functions in FormatManager? Like for example `FillableExtensions()` and `Fillable()`.

Actually, there's no good reason. I looked at how it was implemented in other examples and did the same. If we need an array of action data in the future, we have a ready method.

**Review by aleksandrfedorov97 — APPROVED** [review]

File: web/documentserver-example/csharp/settings.config [hunk 4]
```diff
@@ -8,9 +8,6 @@
   <add key="filename-max" value="50"/>
 
   <add key="files.docservice.fillform-docs" value=".docx|.oform"/>
```
Comment (LinneyS) [inline web/documentserver-example/csharp/settings.config:10]: get from json

**Review by LinneyS — CHANGES_REQUESTED** [review]

**Pull request merged**